### PR TITLE
Add Heii On-Call Notification Provider

### PR DESCRIPTION
**Pull request by hevans66**
Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

Add notification provider for [Heii On-Call](https://heiioncall.com/).

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### Setup Notification

image

### Events

| Event | Before | After |
|-------|------|-----|
| `UP` | - |  |
| `DOWN` | - |  |
| Certificate-expiry | - |  |
| Testing | - |  |

### CODE REVIEW

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Mostly, this is looking good.
I have left some inline remarks about UX and code-quality.

Please attach screenshots of the 3 types of events `DOWN`/`UP`, `Certificate Expiry`[^1] and `Testing` to the description.
(we should really document this one in the contribution guide...)

[^1]: See https://badssl.com/ for broken certs

File: src/components/notifications/HeiiOnCall.vue
Comment (hevans66) [reply src/components/notifications/HeiiOnCall.vue:12]: Not a stupid question at all. I feel like this is one of those that can go either way. I suppose in this context we are using Trigger as basically a proper noun. I'll remove the tranlation.

File: server/notification-providers/heii-oncall.js
Comment (hevans66) [reply server/notification-providers/heii-oncall.js]: I believe my changes do what you asked for. Resolving this comment, but let me know if what I did missed the mark.

File: src/components/notifications/HeiiOnCall.vue
Comment (hevans66) [reply src/components/notifications/HeiiOnCall.vue]: Trigger ID is very front and center as soon you create a trigger in Heii On-Call. It is mentioned in many places on Manual Trigger type documentation. I don't think there will be any confusion as to where to grab that from. If you would like I can add more text here to be explicit about how to find the Trigger ID, but I figured we wanted to keep the Setup Notification screen as clean as possible.

**Comment by hevans66** [comment]
@CommanderStorm Thanks for the review. I believe I have addressed all your concerns.

File: src/components/notifications/HeiiOnCall.vue
Comment (CommanderStorm) [reply src/components/notifications/HeiiOnCall.vue]: As long as how would be clear to a new user of said software, this is fine by me ^^

File: server/notification-providers/heii-oncall.js
Comment (CommanderStorm) [inline server/notification-providers/heii-oncall.js:64]: This should refactor the code to be closer to what the other notification providers look like => keeping it more consistent.

```suggestion
    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
        const okMsg = "Sent Successfully.";
        const payload = heartbeatJSON || {};

        const baseURL = await setting("primaryBaseURL");
        if (baseURL && monitorJSON) {
            payload["url"] = baseURL + getMonitorRelativeURL(monitorJSON.id);
        }

        const config = {
            headers: {
                Accept: "application/json",
                "Content-Type": "application/json",
                Authorization: "Bearer " + notification.heiiOnCallApiKey,
            },
        };
        
        const heiiUrl = `https://heiioncall.com/triggers/${notification.heiiOnCallTriggerId}/`;
        
        // docs https://heiioncall.com/docs#manual-triggers
        try {
            if (!heartbeatJSON) {
                // Testing or general notification like certificate expiry
                payload["msg"] = msg;
                await axios.post(heiiUrl + "alert", payload, config);
                return okMsg;
            }

            if (heartbeatJSON.status === DOWN) {
                await axios.post(heiiUrl + "alert", payload, config);
                return okMsg;
            }
            if (heartbeatJSON.status === UP) {
                await axios.post(heiiUrl + "resolve",payload, config);
                return okMsg;
            }
        } catch (error) {
            this.throwGeneralAxiosError(error);
        }
    }
```

File: src/components/notifications/HeiiOnCall.vue
Comment (CommanderStorm) [inline src/components/notifications/HeiiOnCall.vue:21]: I think adding translations in this way might be easier for our translators

```suggestion
        <template #documentation>
            <a href="https://heiioncall.com/docs" target="_blank">{{ $t("documentationOf", ["Heii On-Call"]) }}</a>
        </template>
```

File: src/lang/en.json
Comment (CommanderStorm) [inline src/lang/en.json:887]:
```suggestion
    "wayToGetHeiiOnCallDetails": "How to get the Tigger ID and API Keys is explained in the {documentation}",
```

File: src/components/notifications/HeiiOnCall.vue
Comment (CommanderStorm) [reply src/components/notifications/HeiiOnCall.vue]: resolving this one, as below is better to track this (below has the fixed indentation ^^)

File: server/notification-providers/heii-oncall.js
Comment (CommanderStorm) [inline server/notification-providers/heii-oncall.js:28]:
```suggestion

```

File: server/notification-providers/heii-oncall.js
Comment (CommanderStorm) [inline server/notification-providers/heii-oncall.js:30]:
```suggestion

```

File: server/notification-providers/heii-oncall.js
Comment (CommanderStorm) [inline server/notification-providers/heii-oncall.js:45]:
```suggestion
                await axios.post(heiiUrl + "resolve", payload, config);
```

**Review by CommanderStorm — COMMENTED** [review]
formatting issues introduced in above fix

**Review by CommanderStorm — APPROVED** [review]
Okay, I changed the formatting of the code a bit to be closer to the other notification providers.
Could you review the changes I added in https://github.com/louislam/uptime-kuma/pull/4485/commits/cc000117ca9b8f888c69a31b9c8f2841b79bab7f?

Other than that, I am happy with the current code.

I am unsure if we should change the translations in this context as suggested in https://github.com/louislam/uptime-kuma/pull/4485#pullrequestreview-1881439791 => I am going to leave this up to you to either accept/resolve

**Comment by hevans66** [comment]
@CommanderStorm awesome! Accepted your changes. Thanks so much for the review.

**Comment by hevans66** [comment]
@CommanderStorm let me know if there is anything else I can do to help this get merged. I'd like to work on a guide for setting up Uptime Kuma with Heii On-Call, and would like to have this in before its published.

**Comment by CommanderStorm** [comment]
Actually there is: 
I missed in the previous review that this is missing the screenshot from the `Testing` state (the "Test"-button in above screenshot)

Other than that I would be happy to merge this into the upcoming v2.0 release.

**Comment by hevans66** [comment]
Testing screenshot has been added.

**Comment by CommanderStorm** [comment]
Thanks for the notification provider!
I think adding it is uncontroversial as it matches the code style and the appropriate tests have been performed.
⇒ merging with junior maintainer approval.

**Pull request merged**
